formula/src/main/java/com/instacart/formula/FormulaRuntime.kt

```diff
@@ -85,75 +86,107 @@ class FormulaRuntime<Input : Any, Output : Any>(
                     effectQueue.addLast(effects)
                 }
 
-                process(isValid)
+                run(isValid)
             }
 
-            val processorManager: FormulaManager<Input, Output> =
-                FormulaManagerImpl(implementation, input, lock, transitionListener)
-
-            manager = processorManager
-
-            process(false)
+            manager = FormulaManagerImpl(implementation, input, transitionListener)
+            forceRun()
             hasInitialFinished = true
 
             lastOutput?.let {
                 onOutput(it)
             }
         } else {
-            process(false)
+            forceRun()
         }
     }
 
+    fun terminate() {
+        manager?.apply {
+            markAsTerminated()
+            performTerminationSideEffects()
+        }
+    }
+
+    private fun forceRun() = run(isValid = false)
+
     /**
-     * Processes the next frame.
+     * Performs the evaluation and execution phases.
+     *
+     * @param isValid Determines if evaluation needs to be run.
```

Comment:
Could this just be called `shouldEvaluate` or something then?

Reply:
Good idea, will rename!